reviewing/dependencies-hierarchy/src/buildDependenciesHierarchy.ts

```diff
@@ -86,16 +86,28 @@ export async function buildDependenciesHierarchy (
     virtualStoreDir: modules?.virtualStoreDir,
     virtualStoreDirMaxLength: modules?.virtualStoreDirMaxLength ?? maybeOpts.virtualStoreDirMaxLength,
   }
-  ; (
-    await Promise.all(projectPaths.map(async (projectPath) => {
-      return [
-        projectPath,
-        await dependenciesHierarchyForPackage(projectPath, currentLockfile, wantedLockfile, opts),
-      ] as [string, DependenciesHierarchy]
-    }))
-  ).forEach(([projectPath, dependenciesHierarchy]) => {
+  // ; (
+  // //   await Promise.all(projectPaths.map(async (projectPath) => {
+  //     return [
+  //       projectPath,
+  //       await dependenciesHierarchyForPackage(projectPath, currentLockfile, wantedLockfile, opts),
+  //     ] as [string, DependenciesHierarchy]
+  //   }))
+  // ).forEach(([projectPath, dependenciesHierarchy]) => {
+  //   result[projectPath] = dependenciesHierarchy
+  // })
```

Comment:
@gluxon I would like to remove the temporary key-value pairs returned from the promise callback by replacing them with direct assignment, like so:

```typescript
  await Promise.all(projectPaths.map(async (projectPath) => {
    result[projectPath] = await dependenciesHierarchyForPackage(projectPath, currentLockfile, wantedLockfile, opts)
  }))
```

The problem is, I'm not sure if there is a reason being the temporary pairs. Since `git blame` shows that you wrote this code, can you explain? If there are no reason, then I would proceed to replace it.

Reply:
I guess it's about determinism?